Entities/Attachable.cpp

```diff
@@ -1,679 +1,450 @@
+				parentAsActor->AlarmPoint(extruded);
+			}
+		}
+
+		return penetrated;
 	}
-}
 
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-//////////////////////////////////////////////////////////////////////////////////////////
-// Method:          TransferJointForces
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Bundles up all the accumulated forces of this Attachable and calcs
-//                  how they transfer to the joint, and therefore to the parent.
+	void Attachable::GibThis(const Vector &impactImpulse, MovableObject *movableObjectToIgnore) {
+		MOSRotating::GibThis(impactImpulse, movableObjectToIgnore);
+		if (m_Parent) {
+			m_Parent->RemoveAttachable(this, true, true);
+		} else {
+			SetParent(nullptr);
+		}
+	}
 
-bool Attachable::TransferJointForces(Vector &jointForces)
-{
-    // Exit trivially if not attahced to anything
-    if (!m_pParent)
-        return false;
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void Attachable::Update() {
+		if (!m_Parent) {
+			m_JointPos = m_Pos + RotateOffset(m_JointOffset);
+		} else {
+			m_PrevPos = m_Pos;
+			m_PrevVel = m_Vel;
+
+			m_JointPos = m_Parent->GetPos() + m_Parent->RotateOffset(GetParentOffset());
+			m_Pos = m_JointPos - RotateOffset(m_JointOffset);
+			m_Vel = m_Parent->GetVel();
+			m_Team = m_Parent->GetTeam();
+			if (InheritsHFlipped() != 0) { m_HFlipped = m_InheritsHFlipped == 1 ? m_Parent->IsHFlipped() : !m_Parent->IsHFlipped(); }
+			if (InheritsRotAngle()) { SetRotAngle(m_Parent->GetRotAngle() + m_InheritedRotAngleOffset); }
+
+			MOSRotating *rootParentAsMOSR = dynamic_cast<MOSRotating *>(GetRootParent());
+			if (rootParentAsMOSR && m_CollidesWithTerrainWhileAttached) {
+				// Note: This safety check exists to ensure the parent's AtomGroup contains this Attachable's Atoms in a subgroup. Hardcoded Attachables need this in order to work, since they're cloned before their parent's AtomGroup exists.
+				if (!rootParentAsMOSR->GetAtomGroup()->ContainsSubGroup(m_AtomSubgroupID)) { AddOrRemoveAtomsFromRootParentAtomGroup(true); }
+
+				float facingAngle = (m_HFlipped ? c_PI : 0) + GetRotAngle() * static_cast<float>(GetFlipFactor());
+				float parentFacingAngle = (m_Parent->IsHFlipped() ? c_PI : 0) + m_Parent->GetRotAngle() * static_cast<float>(m_Parent->GetFlipFactor());
+
+				if (!InheritsRotAngle()) {
+					Matrix atomRotationForSubgroup(facingAngle - parentFacingAngle);
+					Vector atomOffsetForSubgroup;
+					CalculateAtomOffsetForSubgroup(atomOffsetForSubgroup);
+					// Note: This line looks weird because CalculateAtomOffsetForSubgroup already adds JointOffset to the value, so we have to subtract it and added its rotated value instead.
+					atomOffsetForSubgroup += GetJointOffset() - (GetJointOffset() * atomRotationForSubgroup);
```

Comment:
You're right that the comment has that backwards, oops.

It can't be done in CalculateAtomOffsetForSubgroup cause AtomGroup::AddAtoms deals with that such rotation internally. That is to say, it could be done there if AddAtoms was modified not to do that, but I tried to minimize my AtomGroup modifications. I'll put in a TODO to move the work out of AG and do it all here for consistency.